diff --git a/dev-install.sh b/dev-install.sh
index 248efbf..7ea0265 100755
--- a/dev-install.sh
+++ b/dev-install.sh
@@ -323,6 +323,8 @@ cp -af ./$THIRD_PARTY/jquery-linedtextarea.js webroot/assets/jquery-linedtextare
 cp -af ./$THIRD_PARTY/jquery-linedtextarea.css webroot/assets/jquery-linedtextarea/css/jquery-linedtextarea.css
 #End - Copy jquery-linedtextarea from $THIRD_PARTY
 
+npm rebuild node-sass
+
 #Start - Merging All CSS files
 rm -f webroot/css/contrail.unified.css
 #Loop through files in webroot/common/ui/scss and and running through sass compiler and move them to webroot/common/ui/css 
