diff --git a/linux/vr_host_interface.c b/linux/vr_host_interface.c
index 79105f7..69fa8e5 100644
--- a/linux/vr_host_interface.c
+++ b/linux/vr_host_interface.c
@@ -1643,7 +1643,7 @@ linux_if_get_settings(struct vr_interface *vif,
     rtnl_lock();
 
     if (netif_running(dev)) {
-#if (LINUX_VERSION_CODE >= KERNEL_VERSION(4,6,0))
+#if (LINUX_VERSION_CODE >= KERNEL_VERSION(4,6,0) || (LINUX_VERSION_CODE >= KERNEL_VERSION(3,10,0) && LINUX_VERSION_CODE < KERNEL_VERSION(4,0,0)))
         /* ethtool_link_ksettings introduced since kernel 4.6. ethtool_cmd has been removed */
         struct ethtool_link_ksettings ekmd;
         ekmd.base.cmd = ETHTOOL_GSET;
@@ -1651,7 +1651,7 @@ linux_if_get_settings(struct vr_interface *vif,
             settings->vis_speed = ekmd.base.speed;
             settings->vis_duplex = ekmd.base.duplex;
 #endif
-#if (LINUX_VERSION_CODE >= KERNEL_VERSION(3,2,0) && LINUX_VERSION_CODE < KERNEL_VERSION(4,6,0))
+#if ((LINUX_VERSION_CODE >= KERNEL_VERSION(3,2,0) && LINUX_VERSION_CODE < KERNEL_VERSION(3,10,0)) || (LINUX_VERSION_CODE >= KERNEL_VERSION(4,0,0) && LINUX_VERSION_CODE < KERNEL_VERSION(4,6,0)))
         struct ethtool_cmd cmd;
         /* As per lxr, this API was introduced in 3.2.0 */
         if (!(ret = __ethtool_get_settings(dev, &cmd))) {
